pkg/controller/kibana/controller.go

```diff
@@ -111,6 +113,23 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	// Watch for changes to the route
+	/*
+		routePred := predicate.Funcs{
+			UpdateFunc:  func(e event.UpdateEvent) bool { return false },
+			DeleteFunc:  func(e event.DeleteEvent) bool { return handleRoute(e.Meta) },
+			CreateFunc:  func(e event.CreateEvent) bool { return false },
+			GenericFunc: func(e event.GenericEvent) bool { return false },
+		}
+	*/
+	// if err = c.Watch(&source.Kind{Type: &route.Route{}}, &namespacedMapHandler, routePred); err != nil {
+	if err = c.Watch(&source.Kind{Type: &route.Route{}}, &handler.EnqueueRequestForOwner{
+		OwnerType:    &loggingv1.Kibana{},
+		IsController: true,
+	}); err != nil {
```

Comment:
Hi, @ewolinetz , please check if the above is what you meant by "watch the event and enqueue to the owner".

I tested the PR; with this change, deleting route will trigger the reconcile of kibana CR. Thanks.

Reply:
@huikang yes, this is what I had in mind. This should also trigger in the case when an update is performed, right? we may not need this to trigger on a "create" or the "generic" though

Reply:
I think we would also want to add a watch for some of the other object types we create as part of kibana... however we can capture that in a tech debt card so we don't scope creep too much in this fix.

Reply:
> @huikang yes, this is what I had in mind. This should also trigger in the case when an update is performed, right? we may not need this to trigger on a "create" or the "generic" though

Yes, it will trigger reconcile on create / update / delete events. To avoid triggering on a create, shall I add a predicate, which returns false for create event? Or something else.

Reply:
yes, that's all we'll have to do to ignore the event

Reply:
A predicate is added. Please review the change. Thanks.